src/steam/SteamClient.cs

```diff
@@ -115,13 +258,127 @@ public void ShutdownRequested()
     {
     }
 
+    public void WorkshopItemCreated(int result, ulong fileId, bool acceptTermsOfService)
+    {
+        GD.Print("Workshop item create result: ", result, " file: ", fileId, " TOS: ", acceptTermsOfService);
+
+        if (workshopCreateCallback == null)
+        {
+            GD.PrintErr($"Got {nameof(WorkshopItemCreated)} even with no active callbacks");
+            return;
+        }
+
+        var convertedResult = new WorkshopResult
+        {
+            TermsOfServiceSigningRequired = acceptTermsOfService,
+            ItemId = fileId,
+        };
+
+        if (result == Steam.ResultOk)
+        {
+            convertedResult.Success = true;
+        }
+        else
+        {
+            convertedResult.Success = false;
+            convertedResult.TranslatedError = GetDescriptiveSteamError(result);
+        }
+
+        workshopCreateCallback.Invoke(convertedResult);
+        workshopCreateCallback = null;
+    }
+
+    public void WorkshopItemDownloadedLocally(int result, ulong fileId, int appId)
+    {
+        if (appId != AppId)
+            return;
+    }
+
+    public void WorkshopItemInstalledOrUpdatedLocally(int appId, ulong fileId)
+    {
+        GD.Print("Workshop item downloaded or updated, file: ", fileId);
+    }
+
+    public void WorkshopItemDeletedRemotely(int result, ulong fileId)
+    {
+    }
+
+    public void WorkshopItemInfoUpdateFinished(int result, bool acceptTermsOfService)
+    {
+        GD.Print("Workshop item update result: ", result, " TOS: ", acceptTermsOfService);
+
+        if (workshopUpdateCallback == null)
+        {
+            GD.PrintErr($"Got {nameof(WorkshopItemInfoUpdateFinished)} even with no active callbacks");
+            return;
+        }
+
+        var convertedResult = new WorkshopResult
+        {
+            TermsOfServiceSigningRequired = acceptTermsOfService,
+        };
+
+        if (result == Steam.ResultOk)
+        {
+            convertedResult.Success = true;
+        }
+        else
+        {
+            convertedResult.Success = false;
+            convertedResult.TranslatedError = GetDescriptiveSteamError(result);
+        }
+
+        workshopUpdateCallback.Invoke(convertedResult);
+        workshopUpdateCallback = null;
+    }
+
     private void RefreshCurrentUserInfo()
     {
         IsOnline = Steam.LoggedOn();
         SteamId = Steam.GetSteamID();
         IsOwned = Steam.IsSubscribed();
     }
 
+    private string GetDescriptiveSteamError(int result)
+    {
+        // Note: the exact problem varies a bit based on the action being performed, but for faster implementation
+        // these are not separated by operation type (yet)
```

Comment:
```suggestion
        // these are not separated by operation type TODO: (yet)
```
so that it gets detected by rider